src/bin/usbhost.rs

```diff
@@ -0,0 +1,386 @@
+// SPDX-FileCopyrightText: 2024 Foundation Devices, Inc. <hello@foundationdevices.com>
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+#![no_std]
+#![no_main]
+
+extern crate alloc;
+
+use {
+    alloc::vec::Vec,
+    atsama5d27::{
+        aic::{Aic, InterruptEntry, SourceKind},
+        heap::init_heap,
+        logging::init_logging,
+        pio::{self, Pio},
+        pit::Pit,
+        pmc::{PeripheralId, Pmc},
+        tc::Tc,
+        twi::Twi,
+        uart::{Uart, Uart1},
+    },
+    core::{
+        arch::global_asm,
+        panic::PanicInfo,
+        ptr::addr_of_mut,
+        sync::atomic::{compiler_fence, AtomicUsize, Ordering::SeqCst},
+    },
+    log::{debug, error, info, trace, warn},
+};
+
+global_asm!(include_str!("../start.S"));
+
+// MCK: 164MHz
+// Clock frequency is divided by 2 because of the default `h32mxdiv` PMC setting
+const MASTER_CLOCK_SPEED: u32 = 164000000 / 2;
+
+const UHPHS_BASE: usize = 0x00500000;
```

Comment:
This base address should come from the SVDs and generated `utralib` code but it's not there. I've asked @jeandudey to update the SVDs to have the base address included. Before we have it, it's okay to have this address hardcoded.